flux-cli/src/main/java/com/marklogic/flux/tde/JsonTdeBuilder.java

```diff
@@ -108,27 +108,43 @@ private static ObjectNode buildColumnWithRequiredFields(ArrayNode columns, TdeIn
         columnNode.put("scalarType", scalarType);
 
         final String val = (inputs.getColumnVals() != null && inputs.getColumnVals().containsKey(name))
-            ? inputs.getColumnVals().get(name) : name;
+            ? inputs.getColumnVals().get(name) : column.getVal();
```

Comment:
I'm confused by this.
The Column Interface seems to define getVal as an alias for getName. If that's all it is, why do it?
If there's more to it than that, what am I missing?

Reply:
The intent of Column is to separate the construction of the 3 required fields - name/val/scalarType - from the Spark-specific code for doing so. Currently, the Spark-specific code - SparkColumn - doesn't do anything to affect how getVal() works. 

But a future refactor - like possibly today - may results in the Map of columnVals affecting the getVal() method in Column. And Column may also gain getters for each of the other customizable things affected by Maps in WriteStructuredDocumentParams. Not sure yet though, got some other refactors to do first.